hooks/useFetchStakingPoolsData.js

```diff
@@ -81,63 +85,71 @@ const useFetchStakingPoolsData = () => {
+        let totalApr = BigNumber.from(0);
         let tvlInUsd = BigNumber.from(0);
         try {
           // Pool size
           const {
             decoded: { 0: poolBalanceOf },
           } = await stakingPoolsFunctions[stakingPool.id].getBalanceOf.call();
-          poolSize = BigNumber.from(poolBalanceOf)
-          const rewardDataABI = find(MultiRewards, { name: "rewardData" });
-          let method =
-            connexStakingPools[stakingPool.id].rewardsContract.method(
-              rewardDataABI
-            );
-          let res = await method.call(
-            stakingPool.rewardTokens[0].address[VECHAIN_NODE]
-          );
-          const rewardRate = ethers.BigNumber.from(res.decoded.rewardRate);
+          poolSize = BigNumber.from(poolBalanceOf);
+
+          // placeholder for function call return values
+          let method;
+          let res;
+
           const totalSupplyABI = find(IERC20, { name: "totalSupply" });
           method = connex.thor
-            .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
-            .method(totalSupplyABI);
+              .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+              .method(totalSupplyABI);
           res = await method.call();
           const totalLPTokenSupply = BigNumber.from(res.decoded[0]);
           const numberOfLPTokensStaked = poolSize;
 
-          // This is calculated in the amount of vex tokens instead of usd
-          const tokenTvl =
-            poolInfo[stakingPool.id].pair.tokenAmounts[0].toFixed(18);
-
-          // Multiply by two to get the pool TVL as this is a 50-50 pool
-          const tvlInToken = parseUnits(tokenTvl, 18).mul(2);
-
-          apr = rewardRate
-            .mul(NUM_SECONDS_IN_A_YEAR)
-            .mul(parseUnits("1", "ether")) // More units for precision
-            .mul(100) // Convert into percentage
-            // The following two lines take into account
-            // that not all LP tokens are staked on the staking site
-            .mul(totalLPTokenSupply)
-            .div(numberOfLPTokensStaked)
-            // We divide all the rewards by the total reward token TVL
-            .div(tvlInToken);
-
-          tvlInUsd = tvlInToken
-            .mul(parseUnits(poolInfo[stakingPool.id].usdPerToken.toString()))
-            .mul(numberOfLPTokensStaked)
-            .div(totalLPTokenSupply)
-            .div(parseUnits("1", "ether"));
-        } catch (err) {
+          const token0UsdPrice = parseUnits(tokenPrices[poolInfo[stakingPool.id].pair.tokenAmounts[0].token.address].usdPrice.toFixed(18), 18);
+          const token1UsdPrice = parseUnits(tokenPrices[poolInfo[stakingPool.id].pair.tokenAmounts[1].token.address].usdPrice.toFixed(18), 18);
+
+          const token0Amount = parseUnits(poolInfo[stakingPool.id].pair.reserve0.toExact());
+          const token1Amount = parseUnits(poolInfo[stakingPool.id].pair.reserve1.toExact());
+
+          tvlInUsd = token0Amount.mul(token0UsdPrice)
```

Comment:
does this make the assumption that all tokens are 18 decimals? For instance veUSD is 6 decimals, would this fail to handle that case?

Reply:
I just finished an in-depth consideration of this problem. 

For the USD price portion (L109/110), it doesn't matter how many decimal places, because it is only representing USD. 
For L112/113, because `toExact()` gives the decimal representation of the reserve, so `parseUnits` will normalize it to 18 decimal places. 

We should normalize to 18 decimal places instead of having the amounts in accurate decimal places because there is an addition operation when calculating `tvlInUsd`. If we had different decimal places for amounts, the addition would not work.

If we want to talk about limitation, this current implementation will fail when we meet tokens with >18 decimal places. But I don't think we're gonna meet that anytime soon.